File: JobTrackerAPI/Model/Job.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace JobTrackerAPI.Model
{
    public class Job
    {
        [Key, Column(Order = 0)]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int JobID { get; set; }
        [Key, Column(Order = 1)]
        public int JobNumber { get; set; }
        [Key, Column(Order = 2)]
        public string JobTitle { get; set; }
        public string JobLocation { get; set; }
        public string RecruiterName { get; set; }
        public string ? ClientCompanyContactName { get; set; }
        public string RecruiterCompanyName { get; set; }
        public string ClientCompanyName { get; set; }
        public string ? RecruiterPhoneNumber { get; set; }
        public string RecruiterCompanyPhoneNumber { get; set; }
        public string ? ClientCompanyPhoneNumber { get; set; }
        public string RecruiterCompanyLocation { get; set; }
        public string ClientCompanyLocation { get; set; }
        public string ? RecruiterNotes { get; set; }
        public string ? ClientNotes { get; set; }
        public string JobDescription { get; set; }
        public DateTime DateOfSubmission { get; set; }
        public DateTime ? DateOfFollowUp { get; set; }
        public DateTime ? DateOfInterview { get; set; }
        public int ? NotificationID { get; set; }
    }
}

File: JobSearchTracker/JobSearchTracker/src/model/job.ts
import { JTSNotification } from "./notification";

export class JTSJob {
    JobID! :number;
    JobNumber!: number;
    JobTitle!:string;
    JobLocation!:string;
    RecruiterName!:string;
    ClientCompanyContactName?:string;
    RecruiterCompanyName!:string;
    ClientCompanyName!:string;
    RecruiterPhoneNumber?:string;
    RecruiterCompanyPhoneNumber!: string;
    ClientCompanyPhoneNumber?:string;
    RecruiterCompanyLocation!:string;
    ClientCompanyLocation!:string;
    RecruiterNotes?:string;
    ClientNotes?:string;
    JobDescription!:string;
    DateOfSubmission!: Date;
    DateOfFollowUp?: Date;
    DateOfInterview?: Date;
    notification?:JTSNotification;
    notificationID?:number = 0;
}



export class JobEnum {
    id?: number;
    name?: string;
}